src/coreComponents/unitTests/wavePropagationTests/testWavePropagationAdjoint1.cpp

```diff
@@ -0,0 +1,404 @@
+
+  real64 time_n = time;
+  std::cout << "Begin forward:" << time_n << std::endl;
+  // run for 0.25s (100 steps)
+  for( int i=0; i<50; i++ )
+  {
+    rhsForward[i][0]=WaveSolverUtils::evaluateRicker( time_n, *ptrTimeSourceFrequency, *ptrTimeSourceDelay, *ptrRickerOrder );
+    propagator->explicitStepForward( time_n, dt, i, domain, false );
+    time_n += dt;
+  }
+  // cleanup (triggers calculation of the remaining seismograms data points)
+  propagator->cleanup( 1.0, 50, 0, 0, domain );
+
+  // retrieve seismo
+  arrayView2d< real32 > const pReceivers = propagator->getReference< array2d< real32 > >( AcousticWaveEquationSEM::viewKeyStruct::pressureNp1AtReceiversString() ).toView();
+
+  // move it to CPU, if needed
+  pReceivers.move( hostMemorySpace, false );
+
+  // check number of seismos and trace length
+  ASSERT_EQ( pReceivers.size( 1 ), 5 );
+  ASSERT_EQ( pReceivers.size( 0 ), 51 );
+
+  /*----------Save receiver forward----------------------*/
+  array2d< real32 > uForward;
+  uForward.resize( 51, 1 );
+
+  // save receiver value forward on uForward.
+  for( int i = 0; i < 51; i++ )
+  {
+    /*std::cout << "time: " << i*dt  << std::endl;
+       std::cout << "pReceivers1 " << i << ":" << pReceivers[i][0] << std::endl;
+       std::cout << "pReceivers2 " << i << ":" << pReceivers[i][1] << std::endl;
+       std::cout << "pReceivers3 " << i << ":" << pReceivers[i][2] << std::endl;
+       std::cout << "pReceivers4 " << i << ":" << pReceivers[i][3] << std::endl;
+       std::cout << "rhsForward  " << i << ":" << rhsForward[i][0] << std::endl;*/
+    uForward[i][0] = pReceivers[i][0];
+    pReceivers[i][0] = 0.;
+    pReceivers[i][1] = 0.;
+    pReceivers[i][2] = 0.;
+    pReceivers[i][3] = 0.;
+  }
+
+  ASSERT_EQ( rhsForward.size( 1 ), 1 );
+  ASSERT_EQ( rhsForward.size( 0 ), 51 );
+
+  arrayView2d< localIndex > const rNodeIds = propagator->getReference< array2d< localIndex > >( AcousticWaveEquationSEM::viewKeyStruct::receiverNodeIdsString() ).toView();
+  rNodeIds.move( hostMemorySpace, false );
+  localIndex sNodesIdsAfterModif=rNodeIds[0][0];
+  std::cout << "ref back sNodeIds[0][0]:" << sNodesIdsAfterModif << std::endl;
+
+  /*---------------------------------------------------*/
+
+  std::cout << "Begin backward:" << time_n << std::endl;
+
+  //----------Switch source and receiver1 position for backward----------------------//
+  arrayView2d< real64 > const sCoord = propagator->getReference< array2d< real64 > >( AcousticWaveEquationSEM::viewKeyStruct::sourceCoordinatesString() ).toView();
+  arrayView2d< real64 > const rCoord = propagator->getReference< array2d< real64 > >( AcousticWaveEquationSEM::viewKeyStruct::receiverCoordinatesString() ).toView();
+
+  for( int i = 0; i < 3; i++ )
+  {
+    real64 tmp_double;
+    tmp_double=rCoord[0][i];
+    rCoord[0][i]=sCoord[0][i];
+    sCoord[0][i]=tmp_double;
+  }
+
+  sCoord.registerTouch( hostMemorySpace );
```

Comment:
Typically we don't want or need to explicitly call `registerTouch`. If you want to make sure they're on the CPU why not call `move`?